src/axom/primal/operators/evaluate_integral.hpp

```diff
@@ -154,8 +180,8 @@ double evaluate_line_integral(const primal::BezierCurve<double, 2>& c,
  * \param [in] npts_P the number of quadrature points to evaluate the antiderivative
  * \return the value of the integral
  */
-template <class Lambda>
-double evaluate_area_integral(const axom::Array<primal::BezierCurve<double, 2>>& cs,
+template <class Lambda, typename T, int NDIMS>
+double evaluate_area_integral(const axom::Array<primal::BezierCurve<T, NDIMS>>& cs,
```

Comment:
Some quick gut checks on this change -- 
* Does `evaluate_area_integral` work on `BezierCurve`s in 3D?
* Are there restrictions? e.g. that the polygon needs to be planar?
* Should this take a `CurvedPolygon` instead of an array of `BezierCurve`s?
* Should the function below be generalized similarly?

```cpp
template <class Lambda> 
double evaluate_area_integral(const primal::CurvedPolygon<double, 2> cpoly,
                              Lambda&& integrand,
                              int npts_Q,
                              int npts_P = 0)
```

Reply:
Ah I see, I thought I had removed the `evaluate_area_integral( Array<BezierCurve> )` overload, but it slipped past somehow. There is another function down the line that actually does evaluate on `CurvedPolygon` objects how you describe here. 

In general, this method should only be applied to 2D curved polygons, since that's what David's paper describes. It should be able to work on 3D, planar curved polygons as well, but the formulas would have to be a little different for math reasons, namely that the ultimate set of quadrature nodes are axis aligned, and it isn't clear where that axis should be in 3D.

Reply:
I had a comment here, but I think I deleted it somehow. Here was the gist of it:

Oh yes, this function shouldn't be there at all anymore. I thought I had removed the overload for thing function for `Array<BezierCurve>` input, but it somehow slipped past me. There's a function later on that operates directly on curved polygons that has the behavior you're describing, being that it only operates in 2 dimensions.

In general, these algorithms should only work in 2D. They would work on 3D planar curved polygons in principle, but the formulas get a little murkier. David's paper puts the quadrature nodes on the shape aligned with the axes, and for 3D planar shapes, it isn't clear where that axis would be right away. Unlike the other methods that generalized immediately, this one would require a more sophisticated change of variables to get the nodes in the right place.